client/allocrunner/taskrunner/sids_hook.go

```diff
@@ -0,0 +1,245 @@
+package taskrunner
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/allocrunner/interfaces"
+	ti "github.com/hashicorp/nomad/client/allocrunner/taskrunner/interfaces"
+	"github.com/hashicorp/nomad/client/consul"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/pkg/errors"
+)
+
+const (
+	// the name of this hook, used in logs
+	sidsHookName = "consul_sids"
```

Comment:
Maybe something like this? idk `sids` just doesn't seem meaningful in logs, but I suppose log lines will add more context.

```suggestion
	sidsHookName = "consul_si_token"
```